server/src/tree-sitter/analyzer.ts

```diff
@@ -707,62 +733,133 @@ export default class Analyzer {
+      symbolArray.forEach((symbol) => {
+        if (symbol.kind === SymbolKind.Variable) {
+          const sameSymbolInScanResultDocument = scanResultDocSymbols.find((scanResultDocumentSymbol) => {
+            return this.symbolsAreTheSame(scanResultDocumentSymbol, symbol)
+          })
+          if (sameSymbolInScanResultDocument !== undefined) {
+            // Replace final values
+            symbol.finalValue = sameSymbolInScanResultDocument.finalValue
+
+            const commentsForTheSameSymbol = scanResultDocGlobalDeclarationComments[symbol.name].find(item => this.symbolsAreTheSame(item.symbolInfo, symbol))?.comments
+
+            // Extract the modification history of the symbol
+            if (commentsForTheSameSymbol !== undefined) {
+              commentsForTheSameSymbol.forEach((comment) => {
+                /**
+                 *  Example:
+                 *  #   set /home/projects/poky/meta/conf/bitbake.conf:396
+                    #     [_defaultval] "gnu"
+                    TC_CXX_RUNTIME="gnu"
+                 */
+                const regex = /(?<=#\s{3}set\??\s)(?<filePath>\/.*):(?<lineNumber>\d+)/g
+                for (const match of comment.matchAll(regex)) {
+                  const filePath = match.groups?.filePath
+                  const lineNumber = match.groups?.lineNumber
+                  if (filePath !== undefined && lineNumber !== undefined) {
+                    const uri = 'file://' + filePath
+                    const location = Location.create(uri, { start: { line: parseInt(lineNumber) - 1, character: 0 }, end: { line: parseInt(lineNumber) - 1, character: symbol.name.length } })
+                    if (symbol.history === undefined) {
+                      symbol.history = []
+                    }
+                    symbol.history.push(location)
+                  }
+                }
+              })
+            }
 
-    if (originalSymbols.length > 0) {
-      logger.debug(`symbol length for original document: ${originalSymbols.length}`)
-    }
+            if (processedSymbolInformation[symbol.name] === undefined) {
+              processedSymbolInformation[symbol.name] = []
+            }
 
-    // TODO: Further parsing of the scan results using both documents
-    return undefined
+            (processedSymbolInformation[symbol.name] as BitbakeSymbolInformation[]).push(symbol)
+          }
+        }
+      })
+    })
+
+    /**
+     TODO: Achieve the following goals using the scan results
+       [check] 1. Get the final values for variables and show them in the hover
+       [check] 2. Get the complete history of assignments for variables and use them in the go-to-definition feature
+       3. Get the accurate definition for require/inherit/include directives
+       4. Get the accurate recipe includes in the bitbake recipe view in the left panel
+       5. Get the definitions for the overrides (They should exist in the .conf files which are listed in the scan results)
+     */
```

Comment:
You could add `6. Replace parseOnSave behaviour to scan the recipes instead`. Or even as `3.` ;)

Reply:
I will note it in the tracking ticket, I think I will let this PR be dedicated to 1&2